packages/server/postgres/utils/checkEqBase.ts

```diff
@@ -45,7 +46,15 @@ function getPairNeFields(
   for (const [f, defaultValue] of Object.entries(maybeUndefinedFieldsDefaultValues)) {
     const [rethinkValue, pgValue] = [rethinkRow[f], pgRow[f]]
     if (rethinkValue !== undefined) {
-      if (!areEqual(rethinkValue, pgValue)) {
+      if (
+        rethinkValue === null &&
+        maybeNullFieldsDefaultValues &&
+        maybeNullFieldsDefaultValues[f] !== undefined
+      ) {
+        if (!areEqual(maybeNullFieldsDefaultValues[f], pgValue)) {
+          neFields.push(f)
+        }
+      } else if (!areEqual(rethinkValue, pgValue)) {
         neFields.push(f)
       }
     } else {
```

Comment:
nitpick: it feels like this conditional statement is getting a bit unwieldy and hard to parse. what if we did something like:

```
if (rethinkValue === undefined) {
  # logic for when rethinkValue is undefined
} else if (rethinkValue === null && maybeNullFieldDefaultValue) {
  # logic for when rethinkValue is null and maybeNullFieldDefaultValue exists
} else # rethinkValue is defined and non-null, or null but !maybeNullFieldDefaultValue {
  # logic for when rethinkValue is defined and non-null, or null but !maybeNullFieldDefaultValue
}
```

Reply:
I rewrite this part from the other side: "under what circumstance should we consider the two equal (hence continue to the next field)?"

Reply:
> > undefined in RethinkDB === null in PG
> > null in RethinkDB === '' in PG
> > I think this totally makes sense for string values. Do you think this edge case would ever come up with non char fields, such as int for example? i feel like it's theoretically possible, but don't know if it will be a thing in practical usage in our code 🤔
> 
> We can just add expected value in:
> 
> ```ts
> const maybeNullFieldsDefaultValues: {[Property in keyof Partial<User>]: any} = {  
>  overLimitCopy: '' ,
>  intValField: 0
> }  
> ```
> 
> Not sure if this is what you were thinking?

yes i was worried about 0 still being a value whereas null is the absence of a value. but you can say the exact same thing for '' anyway so it's a nonstarter. i can't think of any better alternative, so probably the best we can do is just deal with any edge cases as they come up from this handling.